apis/python/src/tiledbsoma/options/_soma_tiledb_context.py

```diff
@@ -49,54 +63,11 @@ class SOMATileDBContext:
     when *each* object is opened.
     """
 
-    timestamp_start: int = 0
-    """
-    Timestamp range start for operations on SOMA objects. This is usually zero
-    except for specific, unusual query requirements. This has no effect on
-    timestamps used in writing.
-    """
-
-    @timestamp.validator
-    def _validate_timestamps(self, _: Any, __: Any) -> None:
-        if self.timestamp is None:
-            if self.timestamp_start:
-                raise ValueError(
-                    "SOMATileDBContext: if the current `timestamp` is None,"
-                    " the `timestamp_start` cannot be set."
-                )
-        elif not 0 <= self.timestamp_start <= self.timestamp:
-            raise ValueError("SOMATileDBContext: invalid read timestamp range")
-
-    _group_tiledb_ctx_: Optional[tiledb.Ctx] = attrs.field(init=False, default=None)
-    """Cache for the context used to open Groups."""
-
     @property
-    def _group_tiledb_ctx(self) -> tiledb.Ctx:
-        """Internal-only context specifically for Group operations.
-
-        Unlike Arrays, Groups do not take a ``timestamp`` argument and need
-        their timestamps set in their context. This should go away if/when
-        :class:`tiledb.Group` starts taking a timestamp argument like
-        :func:`tiledb.open`.
-        """
-        if self._group_tiledb_ctx_:
-            return self._group_tiledb_ctx_
-        if self.timestamp is None:
-            ctx = self.tiledb_ctx
-        else:
-            ctx = group_timestamp_ctx(
-                self.tiledb_ctx,
-                timestamp_start=self.timestamp_start,
-                timestamp=self.timestamp,
-            )
-        object.__setattr__(self, "_group_tiledb_ctx_", ctx)
-        return ctx
-
-    def _timestamp_arg(self) -> Optional[Tuple[int, int]]:
-        """The value to use as the ``timestamp`` argument to ``tiledb.open``."""
-        if self.timestamp is None:
+    def timestamp(self) -> Optional[datetime.datetime]:
```

Comment:
Is this method necessary or just a convenience for users?

Reply:
It is not strictly necessary. It's here to parallel the structure of `timestamp`/`timestamp_ms` in `TileDBObject`s.